plenum/test/consensus/order_service/test_sim_ordering_node_txn.py

```diff
@@ -0,0 +1,42 @@
+from functools import partial
+import random
+
+import pytest
+
+from plenum.common.constants import VALIDATOR, POOL_LEDGER_ID
+from plenum.common.timer import RepeatingTimer
+from plenum.test.consensus.order_service.sim_helper import order_requests, \
+    get_pools_ledger_size, setup_pool, check_batch_count
+from plenum.test.greek import Greeks
+from plenum.test.node_reg_handler.test_node_reg_handler import build_node_req
+from plenum.test.simulation.sim_random import DefaultSimRandom
+
+
+MAX_BATCH_SIZE = 1
+
+
+@pytest.fixture(params=['promote'])
+def req(request):
+    def wrap(pool_size):
+        if request.param == 'promote':
+            return build_node_req(Greeks[pool_size][0], [VALIDATOR])
+        else:
+            node_index = random.randint(0, pool_size - 1)
+            return build_node_req(Greeks[node_index][0], [])
+    return wrap
+
+
+def test_node_txn_add_new_node(req):
```

Comment:
I think it's very important to have a test where we do oredering of domain and pool transactions ANS do view chanegs in between.